lib/services/task_list_service.dart

```diff
@@ -23,141 +27,340 @@ class TaskListService with LogMixin, ChangeCallbackProvider {
     this._syncService,
   );
 
-  Future<List<Task>> get tasks async {
-    return (await _taskListCrdt).values;
-  }
+  Future<Iterable<TaskRecord>> get allTaskRecords async =>
+      _decodeTasks(await _readAndDecodeTaskListCollectionCrdt()).map(
+        (v) => TaskRecord(
+          v.taskRecord.value,
+          v.taskRecord.clock.node,
+          DateTime.fromMillisecondsSinceEpoch(v.taskRecord.clock.timestamp),
+          v.taskListId,
+        ),
+      );
 
-  Future<List<Task>> getTasksForList(TaskList taskList) async {
-    var sorted = <Task>[];
+  Future<Iterable<Task>> get allTasks async =>
+      (await taskLists).map((taskList) => taskList.elements).expand((e) => e);
 
-    List allTasks = await tasks;
-    for (var i = 0; i < allTasks.length; i++) {
-      if (allTasks[i].taskListID == taskList.id) {
-        sorted.add(allTasks[i]);
-      }
-    }
+  Future<Iterable<TaskList>> get taskLists async => (await _taskListRecords)
+      .where((record) => !record.isDeleted)
+      .map((record) => record.value!);
+
+  Future<TaskList?> getTaskListById(String taskListId) async =>
+      (await _taskListRecordMap)[taskListId]?.value;
+
+  Future<Iterable<Task>> getTasksFromList(String taskListId) async =>
+      getTaskListById(taskListId)
+          .then((taskList) => taskList != null ? taskList.elements : []);
+
+  Future<Iterable<Record<TaskList>>> get _taskListRecords async =>
+      (await _taskListRecordMap).values;
 
-    switch (taskList.sortBy) {
```

Comment:
This was the sorting logic 😄 Did you implement this anywhere else? Otherwise this might have broken the now missing sort functionality. 🙃